input/system/azure/system.go

```diff
@@ -0,0 +1,302 @@
+						SKUName:               util.StringPtrToString(v.SKU.Name),
+						AvailabilityZone:      util.StringPtrToString(v.Properties.AvailabilityZone),
+						StorageGB:             util.Int32PtrToInt(v.Properties.Storage.StorageSizeGB),
+						HighAvailabilityMode:  util.StringCustomTypePtrToString(v.Properties.HighAvailability.Mode),
+						HighAvailabilityState: util.StringCustomTypePtrToString(v.Properties.HighAvailability.State),
+						ReplicationRole:       util.StringCustomTypePtrToString(v.Properties.ReplicationRole),
+					}
+					resourceID = *v.ID
+					break
+				}
+			}
+		}
+	}
+
+	// Server info: Cosmos DB (when server is not found within Flexible Server)
+	if resourceID == "" {
+		clientFactory, err := armcosmosforpostgresql.NewClientFactory(config.AzureSubscriptionID, credential, nil)
+		if err != nil {
+			server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a Cosmos DB client: %v", err)
+			logger.PrintError("Azure/System: Failed to make a Cosmos DB client: %v\n", err)
+			return
+		}
+		// Search a server from the list
+		pager := clientFactory.NewClustersClient().NewListPager(nil)
+		for pager.More() {
+			page, err := pager.NextPage(ctx)
+			if err != nil {
+				logger.PrintError("Azure/System: Failed to advance page of Cosmos DB cluster list: %v\n", err)
+				break
+			}
+			for _, v := range page.Value {
+				if v.ID != nil {
+					resourceIDParts := strings.Split(*v.ID, "/")
+					resourceGroup := resourceIDParts[4]
+					resourceType := resourceIDParts[7]
+					resourceName := resourceIDParts[8]
+
+					if config.AzureDbServerName == resourceName {
+						system.Info.Azure = &state.SystemInfoAzure{
+							Location:                 util.StringPtrToString(v.Location),
+							CreatedAt:                util.TimePtrToTime(v.SystemData.CreatedAt),
+							State:                    util.StringCustomTypePtrToString(v.Properties.State),
+							SubscriptionID:           config.AzureSubscriptionID,
+							ResourceGroup:            resourceGroup,
+							ResourceType:             resourceType,
+							ResourceName:             resourceName,
+							CitusVersion:             util.StringPtrToString(v.Properties.CitusVersion),
+							HighAvailabilityEnabled:  util.BoolPtrToBool(v.Properties.EnableHa),
+							CoordinatorStorageMB:     util.Int32PtrToInt(v.Properties.CoordinatorStorageQuotaInMb),
+							NodeStorageMB:            util.Int32PtrToInt(v.Properties.NodeStorageQuotaInMb),
+							CoordinatorVCores:        util.Int32PtrToInt(v.Properties.CoordinatorVCores),
+							NodeVCores:               util.Int32PtrToInt(v.Properties.NodeVCores),
+							CoordinatorServerEdition: util.StringPtrToString(v.Properties.CoordinatorServerEdition),
+							NodeServerEdition:        util.StringPtrToString(v.Properties.NodeServerEdition),
+							NodeCount:                util.Int32PtrToInt(v.Properties.NodeCount),
+						}
+						resourceID = *v.ID
+						break
+					}
+				}
+			}
+		}
+	}
```

Comment:
Hmm. There's no way to get info about these directly by resource name? We have to iterate? It's probably not a big deal in most situations, but if someone is monitoring 100 servers in Azure, we'll make 100 (at least, subject to pagination) API calls every time we're collecting system stats, no? Due to our snapshot scheduling, these are also going to be happening at basically the same time, so if Azure has burst-unfriendly API limits, that could cause problems.

Reply:
> if someone is monitoring 100 servers in Azure, we'll make 100 (at least, subject to pagination) API calls every time we're collecting system stats, no?

Do you mean if someone has 100 servers in Azure and even if they're monitoring one server in pganalyze, it'll make 100 API calls? Or, do you mean that if someone has 100 servers in Azure and they monitor 100 servers also in pganalyze, it'll make 100 API calls?

For the former, no, it won't, this part of API call should be one per the server that the collector is monitoring. Even with the pager, it'll get all 100 servers info with one API request (fun fact, we can't even specify the pager detail anyways, so we can only get all of them at once). The latter, this is what I covered in https://github.com/pganalyze/collector/pull/591/files#r1735845140. We want to reduce the number of API calls for this indeed, and it would be nice to actually do such thing.
Summarizing, the number of API calls that we will make per server will be up to 3:

1. list all Flexible servers
2. list all Cosmos DB for Postgres servers (only if the server can't be found in 1)
3. get metrics of the server

There is another problem of getting all servers info with one API request, like this: https://github.com/Azure/azure-sdk-for-go/issues/20470
If they indeed have 100 servers, while it's smaller than 30k+ resources like mentioned above, it is still not a small amount of data, so it could cause some memory spike (and compute power to go over that).

Reply:
Soon we'll start tracking system stats every 1 minute, so it's worth thinking about how to reduce API requests. One option might be to cache the API response in a separate data structure and only refresh it every 10 minutes.

Reply:
Agreed, when we increase a system stats collection frequency, we will need to rethink this, and also for other systems too (I believe we have the same problem for all other systems too).
For reducing step 1 and 2 requests, as Sean mentioned, we could only do this once 10 mins and do only 3 every 1 min as a starter. Doing 1 and 2 at the collector level as mentioned in the other comment is also good when the collector is monitoring many servers (likely 3 can't be done at the collector level though).